Objects/typeobject.c

```diff
@@ -1189,8 +1186,10 @@ subtype_dealloc(PyObject *self)
         assert(basedealloc);
         basedealloc(self);
 
-        /* Can't reference self beyond this point */
-        Py_DECREF(type);
+       /* Only decref if the base type is not already a heap allocated type.
+          Otherwise, basedealloc should have decref'd it already */
+        if (type->tp_flags & Py_TPFLAGS_HEAPTYPE && !(base->tp_flags & Py_TPFLAGS_HEAPTYPE))
+                                                Py_DECREF(type);
```

Comment:
odd looking indent, at least via github

Reply:
You are right that was definitely non-intentional. I think my auto rebase messed this up. Anyways, fixed!